### DISCUSSION TITLE
[TINET] ウィンドウスケール・オプションの対応について

### DISCUSSION THREAD

**Discussion by meikyoma** [toppers/users-forum#147] [Q&A]
TINETご担当者様

TINET 1.7 を活用させていただいていますが，ウィンドウスケール・オプション（RFC1323）が未サポートのように見受けられます。
実際に未サポートという認識で合っていますでしょうか？
そうである場合，未サポートに至った理由や技術的な課題などがありましたら，ご教示いただくことは可能でしょうか？

背景としては，
弊社ではTINETのほかに自社製のプロトコルスタックを使用しており，こちらもウィンドウスケール・オプションをサポートしていませんでした。通信相手が当該オプションをサポートし，かつウィンドウサイズ値にとても小さい値が含まれていて，実際にはウィンドウスケール値を乗じた値を使用するべきところをウィンドウサイズ値のみを参照した場合，動作が不安定になるという事例が見つかりました。現在，当該オプションの実装方法を検討しています。
TINET を活用している装置でも同様の課題があると考えていますし，またどのように対応するべきか参考になる情報があれば助かりますので，ご教示いただければ幸いに存じます。

どうぞよろしくお願いいたします。

**Comment by h7ga40** [comment]
TOPPERSホームネットワークWGの長島と申します。

お問い合わせいただいた件について、TINETの設計者に確認いたしましたので、回答をお伝えいたします。

ウィンドウスケール・オプションについては、当初より未サポートとなっております。
これはTINETの設計当時、ターゲットとするデバイスのメモリやCPU処理能力に懸念があったためです。
その後も、ウィンドウスケール・オプションのサポート追加について検討を行わなかったため、現段階でも未サポートのままです。

貴重な事例をご共有いただきありがとうございます。
しかしながら、現時点で対応方法について具体的にお伝えできる情報はありません。
TINETの保守は行っておりますが、RFC1323をTINETに盛り込む検討内容をすぐに提供することは難しい状況です。
ご理解いただければ幸いです。

何卒ご容赦のほどよろしくお願いいたします。

**Comment by meikyoma** [comment]
長島様
ご回答ありがとうございました。
状況を理解できました。感謝いたします。
